### PR TITLE
Clarify and correct Android Sections in WORKSPACE

### PR DESCRIPTION
**Pull request by jubjamie**
Build Tools version updated to latest version shipping with Android Studio.
Added some comments to clarify both this and NDK Level details that have been causing confusion.
Both of these have caused problems within Android studio so these edits should avoid the build tools error.

### CODE REVIEW

**Comment by jubjamie** [comment]
I signed it!

**Comment by jubjamie** [comment]
Added commit for build.gradle to match the build tools version change.

**Comment by petewarden** [comment]
Andrew, can you take a look at this? Thanks!

File: WORKSPACE [hunk 1]
```diff
@@ -29,6 +30,7 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #    name="androidndk",
 #    path="<PATH_TO_NDK>",
 #    api_level=14)
+#Note that API level is not the same as NDK version.
```
Comment (andrewharp) [inline WORKSPACE]: Can you sink to the latest version please and move this line above the newly added NDK comment? It's also missing a space after the #

File: WORKSPACE [hunk 2]
```diff
@@ -20,7 +20,8 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #android_sdk_repository(
 #    name = "androidsdk",
 #    api_level = 23,
-#    build_tools_version = "25.0.1",
+#    build_tools_version = "25.0.2",
+#    #Check your build tools version. Android Studio is shipping with 25.0.2 currently.
```
Comment (andrewharp) [inline WORKSPACE]: Is this correct? I see 25.0.4 in Android Studio now.

In any case I would:
1 ) move the comment above the build_tools_version line
2) rephrase it to say "# Ensure that you have a matching version of build_tools installed, as the SDK default changes periodically."

**Review by andrewharp — CHANGES_REQUESTED** [review]
The build_tools_version also need to be updated in 
tensorflow/tools/ci_build/Dockerfile.android and tensorflow/contrib/android/cmake/build.gradle, thanks

File: WORKSPACE (hunk 1 above)
Comment (jubjamie) [inline WORKSPACE]: Sync to the latest version of what sorry? Tf? 1.1?
I will look at your comments on Monday and I'll get back to it then. I can also make similar edits to the previously mentioned files too if you want?

Comment (andrewharp) [inline WORKSPACE]: Yes, TF master. I made some edits to this file internally to add comments here as well, and those changes have been pushed since you made your edits.

File: WORKSPACE (hunk 2 above)
Comment (jubjamie) [inline WORKSPACE]: After updating I only see 25.0.2 in AS within SDK manager and within my files on this ubuntu version of AS. I like your rephrasing as it allows for those differences.

File: WORKSPACE (hunk 1 above)
Comment (jubjamie) [inline WORKSPACE]: I have made those changes.

**Comment by jubjamie** [comment]
@andrewharp Those changes have been made however I am wondering whether it will break people's production environments? As a new starter with a fresh copy of android studio, I have one version of build_tools but others may still be on a previous version and this change may cause issues? This works both ways though as otherwise it breaks fresh builds on fresh android studio version. Although, curiously, you see a different build_tools_version to me and I've tried to update it! Thoughts?

File: WORKSPACE [hunk 3]
```diff
@@ -29,7 +30,9 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #android_ndk_repository(
 #    name="androidndk",
 #    path="<PATH_TO_NDK>",
-#    api_level=14) # This needs to be 14 or higher to compile TensorFlow.
+#    # This needs to be 14 or higher to compile TensorFlow. Note NDK version is not the API level.
```
Comment (andrewharp) [inline WORKSPACE]: s/Note/Note that/

File: WORKSPACE [hunk 4]
```diff
@@ -20,7 +20,8 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #android_sdk_repository(
 #    name = "androidsdk",
 #    api_level = 23,
-#    build_tools_version = "25.0.1",
+#    #Ensure that your build_tools_version (below) the SDK manager version as it updates periodicaly.
```
Comment (andrewharp) [inline WORKSPACE]: \# Ensure that you have the build_tools_version below installed in the SDK manager as it updates periodically.

File: tensorflow/tools/ci_build/Dockerfile.android [hunk 5]
```diff
@@ -29,7 +29,8 @@ RUN mkdir -p ${ANDROID_DEV_HOME}
 ENV ANDROID_SDK_FILENAME tools_r25.2.5-linux.zip
 ENV ANDROID_SDK_URL https://dl.google.com/android/repository/${ANDROID_SDK_FILENAME}
 ENV ANDROID_API_LEVEL 23
-ENV ANDROID_BUILD_TOOLS_VERSION 25.0.1
+#Build Tools Version liable to change.
```
Comment (andrewharp) [inline tensorflow/tools/ci_build/Dockerfile.android]: missing space after #

**Review by jubjamie — COMMENTED** [review]
Changes reviewed and implemented.

File: WORKSPACE [hunk 6]
```diff
@@ -30,7 +30,7 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #android_ndk_repository(
 #    name="androidndk",
 #    path="<PATH_TO_NDK>",
-#    # This needs to be 14 or higher to compile TensorFlow. Note NDK version is not the API level.
+#    # This needs to be 14 or higher to compile TensorFlow. Note that NDK version is not the API level.
```
Comment (andrewharp) [inline WORKSPACE]: Sorry, this should be "Note that the NDK version is not the API level."

File: WORKSPACE [hunk 7]
```diff
@@ -20,7 +20,8 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #android_sdk_repository(
 #    name = "androidsdk",
 #    api_level = 23,
-#    build_tools_version = "25.0.1",
+#    # Ensure that you have the build_tools_version below installed in the SDK manager as it updates periodically.
```
Comment (andrewharp) [inline WORKSPACE]: line wrap at 80 characters

**Review by jubjamie — COMMENTED** [review]
Line Wrap implemented

File: WORKSPACE [hunk 8]
```diff
@@ -29,7 +31,9 @@ load("//tensorflow:workspace.bzl", "tf_workspace")
 #android_ndk_repository(
 #    name="androidndk",
 #    path="<PATH_TO_NDK>",
-#    api_level=14) # This needs to be 14 or higher to compile TensorFlow.
+#    # This needs to be 14 or higher to compile TensorFlow. Note that NDK version is not the API level.
+#    api_level=14)
+
```
Comment (andrewharp) [inline WORKSPACE]: extra blank line and missing "the"

**Review by jubjamie — COMMENTED** [review]
Requested changes made + new line for line wrapping.

**Review by andrewharp — APPROVED** [review]

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please

**Comment by jubjamie** [comment]
Do I need to do anything to address the failed builds? I'm not sure they are failing and, considering how basic the changes were i'm not sure if there's anything I could do!
Thanks

**Comment by yifeif** [comment]
@jubjamie thanks for asking. The failure looks like a flake. Running @tensorflow-jenkins test this please.

**Comment by jubjamie** [comment]
@yifeif Hmmm ok i'll let Jenkins do his thing! Thanks

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please.
